src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -101,6 +101,35 @@ public override bool Execute()
             return true;
         }
 
+        // Filenames with special characters like '#' and'{' get written
+        // into the section names in the resulting .editorconfig file. Later,
+        // when the file is parsed in configuration options these special
+        // characters are interpretted as invalid values and ignored by the
+        // processor. We encode the special characters in these strings
+        // before writing them here.
+
+        private static string EncodeString(string p)
+        {
+            StringBuilder builder = new StringBuilder();
```

Comment:
can you use a PooledStringBuilder here?  thanks!

Reply:
@CyrusNajmabadi Why not even avoid creating StringBuilder at all?
The caller already has a StringBuilder that we can reuse.

Reply:
that would work for me as well.

Reply:
Passed in the `StringBuilder` from the caller.